posydon/popsyn/synthetic_population.py

```diff
@@ -230,6 +231,10 @@ def parse(self, S1_state=None, S2_state=None, binary_state=None,
                     df_sel_met = pd.concat([df_sel_met, df_tmp])
                     del df_tmp
 
+            if k != 0:
```

Comment:
Do you mean `k>0`? Here both statements are equivalent. But I'd say `k>0` is more understood as after the first iteration, while `k!=0` means not in the first iteration.